BEIMA.Backend/MongoService/MongoConnector.cs

```diff
@@ -0,0 +1,192 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using MongoDB.Driver;
+using MongoDB.Bson;
+
+namespace BEIMA.Backend.MongoService
+{
+    public sealed class MongoConnector
+    {
+        //Public class members
+        public ServerType CurrentServerType { get; }
+
+        //Contains instance variables
+        private readonly MongoClient client = null;
+        private static readonly Lazy<MongoConnector> instance = new(() => new MongoConnector());
+
+        //Environment variables
+        private readonly string dbName = Environment.GetEnvironmentVariable("DatabaseName");
+        private readonly string deviceCollection = Environment.GetEnvironmentVariable("DeviceCollectionName");
```

Comment:
I think this would be better to put in a constants class rather than an environment variable?

Reply:
I see it as an environment variable. The user does not have to use "beima" or "devices" as the collection names, it is completely dependent on how they want their environment set up. This is an open source project, so people might use their own name in place of "beima", or they might want to use their own language in place of "devices".

Reply:
The beima one makes sense to me, but I wouldn't think we would want to change the device name? I guess I'm just unfamiliar with this type of database.

Reply:
Collections in MongoDB are equivalent to tables in SQL. If someone speaking a language other than English wanted to change the name of the "devices" collection to whatever "devices" is in their language, they should have to ability to do so. Mongo is very flexible when it comes to these type of things.

Reply:
Ok, as long as it's the standard I'm cool with it.